### DISCUSSION TITLE
Expiring cookie with a configurable timeout

### DISCUSSION THREAD

**Discussion by tpaksu** [tpaksu/laravel-otp-login#18] [General]
@usamaJ17 I've updated the code to include tests, so we don't break anything when implementing that feature. Also, the `otp_timeout` variable was referring to the duration that the OTP code sent will get expired, like, if you don't enter that code in 5 minutes, the code will be obsolete, and you need to request another one from the system. So I changed the default to 5 minutes instead of the number which was written in the config.

So we can discuss the idea of expiring the cookie here.

**Comment by usamaJ17** [comment]
@tpaksu Yes, I understand the idea of otp_timeout. But when we verify OTP its status will remain verify for always, But in some cases where data is sensitive, and also in my case I want user to verify its identity after every 2 hours, I want that after every 2 hours a new OTP will be sent and user has to verify it again inorder to remain in system. So for this I added a new feature of discard_time. i define it to 2 hours and after 2 hours OTP status will be change to discarded from verify, and user will need to verify it again.

**Reply by tpaksu** [reply]
Yes, I understood the idea, just wanted to create a discussion here about the implementation details. 

So can you remind me why setting the cookie with that specific timeout (2h in your case) was not a valid approach? If it was about refreshing the cookie every time the page changes, we can make it work without the updates. Will it solve the problem?
